gcc/rust/checks/errors/privacy/rust-privacy-reporter.cc

```diff
@@ -31,11 +32,61 @@ PrivacyReporter::PrivacyReporter (
     current_module (tl::nullopt)
 {}
 
+// Find a proc_macro, proc_macro_derive or proc_macro_attribute
+// attribute in a vector of attribute
+static tl::optional<std::string>
+find_proc_macro_attribute (const AST::AttrVec &outer_attrs)
+{
+  tl::optional<std::string> result;
+
+  for (auto &a : outer_attrs)
+    {
+      auto &segments = a.get_path ().get_segments ();
+      if (segments.size () != 1)
+	continue;
+      auto name = segments.at (0).get_segment_name ();
+      if (name == "proc_macro" || name == "proc_macro_attribute"
+	  || name == "proc_macro_derive")
+	result = {name};
+    }
+
+  return result;
+}
```

Comment:
...by removing the temp variable and just returning as soon as we see one proc macro attribute
```suggestion
static tl::optional<std::string>
find_proc_macro_attribute (const AST::AttrVec &outer_attrs)
{
  for (auto &a : outer_attrs)
    {
      auto &segments = a.get_path ().get_segments ();
      if (segments.size () != 1)
	continue;
      auto name = segments.at (0).get_segment_name ();
      if (name == "proc_macro" || name == "proc_macro_attribute"
	  || name == "proc_macro_derive")
	return name;;
    }

  return tl::nullopt;
}
```
right?

Reply:
Yes you're completely right

Reply:
Fixed in [d5f8243](https://github.com/Rust-GCC/gccrs/pull/2446/commits/d5f82432f614bf2cd13b7b391fa0c2e379eedd69).